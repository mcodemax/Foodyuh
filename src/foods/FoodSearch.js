import React, { useState, useContext, useEffect } from 'react';
import FoodyuhApi from '../foodyuhApi';
import { Formik, Form, Field, ErrorMessage } from 'formik';
import UserContext from '../auth/UserContext';
import { useNavigate } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import * as Yup from 'yup';
import Food from './Food';

//may need to pass a f(0 in from parent level PlateDetails compoenent to add fdcid)
function FoodSearch({plateId}) {
  const [foods, setFoods] = useState();
  const [errors, setErrors] = useState([]);

  const initialValues = {
    search: '',
  };

  const onSubmit = async (values, { resetForm }) => {

    try {
      const res = await FoodyuhApi.searchForFoods(values.search);
      setFoods(res);
      console.log(res)
    } catch (error) {
      //implment error handling later
        resetForm({});
    }
    // https://github.com/jaredpalmer/formik/issues/446
  };

  const validationSchema = Yup.object({
    search: Yup.string()
      .min(1)
      .max(255, 'Max search query is 255 characters')
      .required('Required'),
  });

  return (
    <div className='FoodSearch'>
      <div className='FoodSearch-formik'>
        <Formik
            initialValues={initialValues}
            validationSchema={validationSchema}
            onSubmit={onSubmit}
          >
            <>
              <p>Search for a Food!</p>
              <Form className={`FoodSearch-form`}>
                <div className={`FoodSearch-form-search`}>
                  <label htmlFor='search' >Search</label>
                  <Field type='text' id='search' name='search' placeholder='Find a food'/>
                  <ErrorMessage name='search' />
                </div>
                <button type='submit' id='submit'>
                  Submit
                </button>
              </Form>
            </>
        </Formik>
      </div>
      <div className='FoodSearch-foodslist'>
        {foods ? foods.map( food => {
              return (
                <>
                  <Food food={food} plateId={plateId} key={uuidv4()}/>
                  {/* pass in f() to food component to add call f() to add food to the plate */}
                  {console.log(food)}
                  {/*
                    let's pass in params to a food card 
                    lets display:
                      brandOwner
                      let's keep secret html value:
                      fdcId
                  */}
                </>
                // add a hyperlink tag to navigate to indiv plate details
              );
            }) : null}
      </div>

    </div>
  )
}

export default FoodSearch;
