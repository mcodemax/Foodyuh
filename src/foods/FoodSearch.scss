.FoodSearch {
  display: flex;
  flex-direction: column;

  .FoodSearch-foodslist {
    margin: 0 auto;
    max-width: 1400px;

    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 10px 20px; /* row-gap column gap */
  }
  button {
    text-decoration: none;
    display: inline-block;
    text-align: center;
    vertical-align: middle;
    width: 100px;
    user-select: none;
    padding: 0.3rem 0.75rem;
    margin: 20px 30px;
    font-size: 1rem;
    border-radius: 0.25rem;
    color: #fff;
    background-color: #007bff;
    border-color: #007bff;
    margin-top: 1rem !important;
    font-weight: 600 !important;
    cursor: pointer;
  }
}
