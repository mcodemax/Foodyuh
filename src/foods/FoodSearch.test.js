import React from "react";
import { render } from "@testing-library/react";
import FoodSearch from "./FoodSearch";

it("renders without crashing", function () {
  render(<FoodSearch />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<FoodSearch />);
  expect(asFragment()).toMatchSnapshot();
});
