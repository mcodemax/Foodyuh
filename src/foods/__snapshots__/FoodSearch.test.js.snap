// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="FoodSearch"
  >
    <div
      class="FoodSearch-formik"
    >
      <p>
        Search and add a Food!
      </p>
      <form
        action="#"
        class="FoodSearch-form"
      >
        <div
          class="FoodSearch-form-search"
        >
          <label
            for="search"
          >
            Search
          </label>
          <input
            id="search"
            name="search"
            placeholder="Find a food"
            type="text"
            value=""
          />
        </div>
        <button
          id="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="FoodSearch-foodslist"
    />
  </div>
</DocumentFragment>
`;
